notebooks/02_analyses/sentiment_analysis.ipynb

```diff
@@ -0,0 +1,776 @@
+   "metadata": {},
+   "source": [
+    "# Monthly Keyword Analysis \n",
+    "\n",
+    "This notebook ingests the preprocessed data from `../interim/text` downloaded by `download_datasets.ipynb` and uses a text classifier from the package`flair` to identify the sentiment for each month. \n",
+    "\n",
+    "Finally, the data is saved as a single csv file and pushed to remote storage for visualization with Superset. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/opt/app-root/lib/python3.6/site-packages/torch/cuda/__init__.py:52: UserWarning: CUDA initialization: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx (Triggered internally at  /pytorch/c10/cuda/CUDAFunctions.cpp:100.)\n",
+      "  return torch._C._cuda_getDeviceCount() > 0\n"
+     ]
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "import os\n",
+    "import datetime\n",
+    "import re\n",
+    "from dotenv import load_dotenv\n",
+    "import seaborn as sns\n",
+    "from collections import Counter\n",
+    "\n",
+    "load_dotenv(\"../../.env\")\n",
+    "\n",
+    "import sys\n",
+    "\n",
+    "sys.path.append(\"../..\")\n",
+    "from src import utils\n",
+    "\n",
+    "from flair.data import Sentence\n",
+    "from flair.models import TextClassifier"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "BASE_PATH = os.getenv(\"LOCAL_DATA_PATH\", \"../../data\")\n",
+    "\n",
+    "LAST_MONTH_DATE = datetime.datetime.now().replace(day=1) - datetime.timedelta(\n",
+    "    days=1\n",
+    ")\n",
+    "year = LAST_MONTH_DATE.year\n",
+    "month = LAST_MONTH_DATE.month"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if os.getenv(\"RUN_IN_AUTOMATION\"):\n",
+    "    df = pd.read_csv(\n",
+    "        f\"{BASE_PATH}/interim/text/fedora-devel-{year}-{month}.mbox.csv\"\n",
+    "    )\n",
+    "    df.head()\n",
+    "\n",
+    "else:\n",
+    "    df = utils.load_dataset(f\"{BASE_PATH}/interim/text/\")\n",
+    "    df.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(44304, 3)"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Text Preprocessing\n",
+    "\n",
+    "Due to the casual nature of email writing, along with some known useless artifacts present in our textual dataset, we need to clean our data a bit before performing our analysis. "
```

Comment:
I think you should rebase this on top of the last merged PR, it moves the data pre-processing out of the analysis notebooks. So you can read in the cleaned data directly.